codelab/JinghangCodeLab/MagicNumberConverterTest.java

```diff
@@ -0,0 +1,16 @@
+package JinghangCodeLab;
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Test;
+
+public class MagicNumberConverterTest {
+    @Test
+    public void testConvertBinaryNumber() {
+        MagicNumberConverter convert = new MagicNumberConverter();
+                assertEquals(-10,convert.binaryToDecimal("-1010"));
```

Comment:
add more unit tests cases to test other possible inputs, for example: "11111", "0", "1".

long string input such as "11111111111111111111".

consider other corner cases such as empty string as input, or when the string is very long, for example, 50 digits, how do you handle such cases?